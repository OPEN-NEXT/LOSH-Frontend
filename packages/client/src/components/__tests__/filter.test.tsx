import * as React from "react";
import ShallowRenderer from "react-test-renderer/shallow"; // ES6
import Filter from "../filter";
const renderer = ShallowRenderer.createRenderer();

jest.mock("antd", () => {
	const Menu = (): null => null;
	Menu.Item = (): null => null;
	return {
		Menu,
		Dropdown: (): null => null,
		Button: (): null => null,
		message: (): null => null,
		Space: (): null => null,
		Typography: {
			Text: (): null => null,
		},
		Select: (): null => null,
	};
});
jest.mock("@ant-design/icons", () => {
	return {
		DownOutlined: (): null => null,
		UserOutlined: (): null => null,
		CheckOutlined: (): null => null,
	};
});
jest.mock("@apollo/client", () => {
	// This should ideally just use Apollo's MockedProvider:
	// https://www.apollographql.com/docs/react/development-testing/testing/
	// Couldn't get it to work with Snapshots though.
	return {
		useQuery: () => ({
			data: {
				organizations: [],
				repos: [],
			},
		}),
		gql: (): null => null,
	};
});

describe("Filter", () => {
	it("renders correctly", () => {
		const tree = renderer.render(
			<Filter filters={{}} onFilterChange={() => null} />
		);
		expect(tree).toMatchSnapshot();
	});
});
