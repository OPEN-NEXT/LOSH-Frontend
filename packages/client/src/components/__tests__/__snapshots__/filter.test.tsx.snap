// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<Space
  className="filter"
  wrap={true}
>
  <Dropdown
    overlay={
      <Menu>
        <Unknown
          icon={false}
          onClick={[Function]}
        >
          Strong Copyleft
        </Unknown>
        <Unknown
          icon={false}
          onClick={[Function]}
        >
          Weak Copyleft
        </Unknown>
        <Unknown
          icon={false}
          onClick={[Function]}
        >
          Permissive
        </Unknown>
      </Menu>
    }
    trigger={
      Array [
        "click",
      ]
    }
  >
    <Button>
      License 
      <DownOutlined />
    </Button>
  </Dropdown>
  <Select
    filterOption={[Function]}
    onChange={[Function]}
    optionFilterProp="children"
    placeholder="Organization"
    showSearch={true}
    style={
      Object {
        "width": 240,
      }
    }
  />
</Space>
`;
